test/k8sT/Services.go

```diff
@@ -729,23 +729,34 @@ var _ = Describe("K8sServicesTest", func() {
 				testCurlRequestFail(testDSClient, httpURL)
 				testCurlRequestFail(testDSClient, tftpURL)
 
-				// Ensure the NodePort cannot be bound from any redirected address
-				failBind(localCiliumHostIPv4, data.Spec.Ports[0].NodePort, "tcp", k8s1NodeName)
-				failBind(localCiliumHostIPv4, data.Spec.Ports[1].NodePort, "udp", k8s1NodeName)
-				failBind("127.0.0.1", data.Spec.Ports[0].NodePort, "tcp", k8s1NodeName)
-				failBind("127.0.0.1", data.Spec.Ports[1].NodePort, "udp", k8s1NodeName)
-				failBind("", data.Spec.Ports[0].NodePort, "tcp", k8s1NodeName)
-				failBind("", data.Spec.Ports[1].NodePort, "udp", k8s1NodeName)
-
-				failBind("::ffff:127.0.0.1", data.Spec.Ports[0].NodePort, "tcp", k8s1NodeName)
-				failBind("::ffff:127.0.0.1", data.Spec.Ports[1].NodePort, "udp", k8s1NodeName)
-				failBind("::ffff:"+localCiliumHostIPv4, data.Spec.Ports[0].NodePort, "tcp", k8s1NodeName)
-				failBind("::ffff:"+localCiliumHostIPv4, data.Spec.Ports[1].NodePort, "udp", k8s1NodeName)
 			}
 
 			wg.Wait()
 		}
 
+		testFailBind := func() {
+			var data v1.Service
+			var localCiliumHostIPv4 string
+
+			err := kubectl.Get(helpers.DefaultNamespace, "service test-nodeport").Unmarshal(&data)
+			Expect(err).Should(BeNil(), "Can not retrieve service")
```

Comment:
I think we should use `ExpectWithOffset(1, ...)` here, as we are moving these Expects one level into call stack.